### PR TITLE
Env: Check for legacy installs and provide the option to delete them.

### LINKED ISSUE #20323: @wordpress/env: Inform user that they can safely delete `../$cwd-wordpress` and `../$cwd-tests-wordpress`
#20002 changed the directory that WordPress is downloaded into from `../$cwd-wordpress` (and `../$cwd-tests-wordpress`) to `~/.wp-env/$hash/WordPress` (and `~/.wp-env/$hash/tests-WordPress`).

Users of `wp-env`, after updating to version 1.0.0, can safely delete these repositories and reclaim about 500 MB of disk space.

We should add a check to `wp-env start` which detects if these directories exist and, if they do, informs the user that they are safe to delete.

cc. @epiqueras @noahtallen

### PR DESCRIPTION
**Pull request by epiqueras**
Closes #20323.

### CODE REVIEW

File: packages/env/lib/env.js [hunk 1]
```diff
@@ -219,6 +222,52 @@ module.exports = {
...
+ * Checks for legacy installs and provides
+ * the user the option to delete them.
+ *
+ * @param {Object} spinner A CLI spinner which indicates progress.
+ */
+async function checkForLegacyInstall( spinner ) {
+	const basename = path.basename( process.cwd() );
+	const installs = [
+		`../${ basename }-wordpress`,
+		`../${ basename }-tests-wordpress`,
```
Comment (noisysocks) [inline packages/env/lib/env.js]: I assume `rimraf` handles the `/` correctly on Windows, but should we output `\` in the confirmation message for Windows users? I suppose it doesn't matter.

File: packages/env/lib/env.js [hunk 2]
```diff
@@ -219,6 +222,52 @@ module.exports = {
...
+	];
+	await Promise.all(
+		installs.map( ( install ) =>
+			fs
+				.access( install )
```
Comment (noisysocks) [inline packages/env/lib/env.js]: I wish `fs.exists` wasn't deprecated—it's so much more readable 😢

**Review by noisysocks — APPROVED** [review]
This works great! 👍

File: packages/env/lib/env.js (hunk 1 above)
Comment (epiqueras) [reply packages/env/lib/env.js]: Is there a convention for this?

File: packages/env/lib/env.js (hunk 2 above)
Comment (epiqueras) [reply packages/env/lib/env.js]: Yeah

File: packages/env/lib/env.js (hunk 1 above)
Comment (noisysocks) [reply packages/env/lib/env.js]: I have no idea. It's okay. I think `../` is common enough that Windows users running `wp-env` would know what it means 😇

Comment (epiqueras) [reply packages/env/lib/env.js]: Yeah, that's what I think.

**Pull request merged**